glfw/src/Window.zig

```diff
@@ -3221,3 +3221,104 @@ test "setScrollCallback" {
         }
     }).callback);
 }
+
+fn failedToCreateWindow(err: Error) void {
+    // return without fail, because most of our CI environments are headless / we cannot open
+    // windows on them.
+    std.debug.print("note: failed to create window: {}\n", .{err});
+    return;
+}
+
+test "hint-attribute default value parity" {
+    try glfw.init(.{});
+    defer glfw.terminate();
+    const window_a = @as(Error!Window, handle: {
+        // manually create window, to avoid call to `Window.Hints.set` in `Window.create`, which
+        // would overwrite the default hint values set by GLFW itself.
+        const handle = c.glfwCreateWindow(640, 480, "Hello, Zig!", null, null);
+        getError() catch |err| break :handle err;
+        break :handle Window.from(handle.?) catch |err| break :handle err;
+    }) catch |err| return failedToCreateWindow(err);
```

Comment:
thought: maybe using a global var `testing_set_no_hints` and `comptime if` inside `Window.create` to disable the hinting calls would be simpler than all of this logic?

Reply:
Are you implying we use a global `comptime var`? Because I don't think that's possible. Or do you mean something like `const testing_set_no_hints = std.meta.globalOption("glfw_testing_set_no_hints") orelse false`? I don't know if that would work, since I don't think (?) tests are treated as "root", though I'll have to check that.

Reply:
A different solution occurred to me. Make a normal global variable, but with this declaration:
```zig
var testing_ignore_hints_struct = if (@import("builtin").is_test) false else @as(void, {});
```
Which would change the code inside `create` to:
```zig
const ignore_hints_struct = if (comptime @import("builtin").is_test) testing_ignore_hints_struct else false;
if (!ignore_hints_struct) {
    try hints.set();
    defer defaultHints() catch unreachable; // this should be unreachable, being that this should be caught in the previous call to `Hints.set`.
}
```

Reply:
I've committed the changes. I'll leave this as unresolved in case you have anything you want to clarify or suggest.

Reply:
Smart approach - I wouldn't have thought of that. Nice work.